src/main/java/com/minecolonies/colony/Field.java

```diff
@@ -0,0 +1,467 @@
+import com.minecolonies.inventory.InventoryField;
+import com.minecolonies.util.BlockPosUtil;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.inventory.Container;
+import net.minecraft.inventory.Slot;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemSeeds;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.BlockPos;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.world.World;
+import net.minecraftforge.common.IPlantable;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Handles the field class.
+ */
+public class Field extends Container
+{
+    /**
+     * Tag to store the location.
+     */
+    private static final String TAG_LOCATION = "location";
+
+    /**
+     * Tag to store if the field has been taken.
+     */
+    private static final String TAG_TAKEN = "taken";
+
+    /**
+     * Tag to store the fields positive length.
+     */
+    private static final String TAG_LENGTH_PLUS = "length+";
+
+    /**
+     * Tag to store the fields positive width.
+     */
+    private static final String TAG_WIDTH_PLUS = "width+";
+
+    /**
+     * Tag to store the fields negative length.
+     */
+    private static final String TAG_LENGTH_MINUS = "length-";
+
+    /**
+     * Tag to store the fields negative width.
+     */
+    private static final String TAG_WIDTH_MINUS = "width-";
+
+    /**
+     * Tag to store the fields stage.
+     */
+    private static final String TAG_STAGE = "stage";
+
+    /**
+     * The max width/length of a field.
+     */
+    private static final int MAX_RANGE = 5;
+
+    /**
+     * The fields location.
+     */
+    private BlockPos location;
+
+    /**
+     * The colony of the field.
+     */
+    private final Colony colony;
+
+    /**
+     * Has the field be taken by any worker?
+     */
+    private boolean taken = false;
+
+    /**
+     * Checks if the field needsWork (Hoeig, Seedings, Farming etc)
+     */
+    private boolean needsWork = false;
+
+    /**
+     * Has the field been planted?
+     */
+    private FieldStage fieldStage = FieldStage.EMPTY;
+
+    /**
+     * The set seed type for the field.
+     */
+    private ItemSeeds seed;
+
+    /**
+     * The length to plus x of the field.
+     */
+    private int lengthPlusX;
+
+    /**
+     * The width to plus z of the seed.
+     */
+    private int widthPlusZ;
+
+    /**
+     * The length to minus xof the field.
+     */
+    private int lengthMinusX;
+
+    /**
+     * The width to minus z of the seed.
+     */
+    private int widthMinusZ;
+
+    /**
+     * The inventorySlot of the field.
+     */
+    private InventoryField inventory;
+
+    /**
+     * Describes the stage the field is in.
+     * Like if it has been hoed, planted or is empty.
+     */
+    public enum FieldStage
+    {
+        EMPTY,
+        HOED,
+        PLANTED
+    }
+
+
+     /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Block comment has incorrect indentation level 5, expected is 4, indentation should be the same level as line 136. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.indentation.CommentsIndentationCheck)